tapgarden/batch.go

```diff
@@ -429,12 +429,6 @@ func (m *MintingBatch) validateUniCommitment(newSeedling Seedling) error {
 				"with the universe commitment flag enabled")
 		}
 
-		if !newSeedling.HasGroupKey() {
```

Comment:
> At the time of validation, the group key may not yet exist and could be
generated later during batch sealing.

How can the group key be generated later on? We need to know the group key upfront to create any of the set of relevant commitments.

Reply:
Previously, the `if !newSeedling.HasGroupKey()` check within `ChainPlanter.prepAssetSeedling` was used to verify that a seedling had its `GroupInfo` set upon being added to a batch. This worked correctly for seedlings requested through `c.seedlingReqs` that already had a group key. However, this check created an issue for seedlings whose group key is generated later, during the batch seal step. In this scenario, the seedlings would not yet have their `GroupInfo` set, causing the `HasGroupKey()` check to fail incorrectly.